lib/Sema/CSDiagnostics.cpp

```diff
@@ -8143,3 +8143,24 @@ bool DefaultExprTypeMismatch::diagnoseAsError() {
 
   return true;
 }
+
+bool MissingExplicitExistentialCoercion::diagnoseAsError() {
+  auto diagnostic = emitDiagnostic(diag::result_requires_explicit_coercion,
+                                   ErasedResultType);
+  fixIt(diagnostic);
+  return true;
+}
+
+bool MissingExplicitExistentialCoercion::diagnoseAsNote() {
+  auto diagnostic = emitDiagnostic(
+      diag::candidate_result_requires_explicit_coercion, ErasedResultType);
+  fixIt(diagnostic);
+  return true;
+}
+
+void MissingExplicitExistentialCoercion::fixIt(
+    InFlightDiagnostic &diagnostic) const {
+  auto printOpts = PrintOptions::forDiagnosticArguments();
+  diagnostic.fixItInsertAfter(getSourceRange().End,
+                              "as " + ErasedResultType->getString(printOpts));
```

Comment:
Could we insert parentheses in cases where the expression being coerced is an argument to a function that can be opened?

Reply:
Sure!

Reply:
Implemented!